CMakeLists.txt

```diff
@@ -42,6 +42,11 @@ include(UseDoxygen)
 # Switch on warnings.
 add_definitions(-Wall -Wextra)
 
+# Forcibly enable C++11. (Adding this flag is easier than using cmake macros.)
+#
+# TODO: move on to C++17.
+add_definitions("-std=c++11")
```

Comment:
This will not produce a correct error if the compiler is not C++11 compatible, will not work with MSVC, and will not work with old versions of GCC (they requires `-std=c++0x`).

I suggest that you [copy the testing code from Aseba](/aseba-community/aseba/blob/master/CMakeLists.txt).